components/checkbox/checkbox.test.js

```diff
@@ -1,520 +1,507 @@
   });
-  beforeEach(() => {});
-
-  // Teardown
-  afterEach(() => {
-    propsData = basePropsData;
-    attrs = {};
-    slots = {};
-    provide = {};
-    listeners = {};
+
+  beforeEach(function () {
+    updateWrapper();
+  });
+
+  afterEach(function () {
+    mockProps = {};
+    mockAttrs = {};
+    mockSlots = {};
+    mockProvide = {};
+    mockListeners = {};
   });
-  afterAll(() => {});
 
   describe('Presentation Tests', () => {
-    describe('When the checkbox renders', () => {
-      // Test Setup
-      beforeEach(() => { _setWrappers(); });
+    it('Should render the component', () => {
+      expect(wrapper).toBeDefined();
+      expect(checkbox).toBeDefined();
+    });
 
-      it('should exist', () => { expect(wrapper.exists()).toBe(true); });
-      it('should have checkbox group class', () => {
-        expect(wrapper.find('.d-checkbox-group').exists()).toBe(true);
+    it('should have checkbox group class', () => {
+      expect(wrapper.find('.d-checkbox-group').exists()).toBe(true);
+    });
+
+    describe('Checkbox Input Tests', () => {
+      it('should exist', () => {
+        expect(input.exists()).toBe(true);
       });
 
-      describe('Checkbox Input Tests', () => {
-        it('should exist', () => { expect(input.exists()).toBe(true); });
-        it(
-          'should have type checkbox',
-          () => { expect(input.attributes('type')).toBe('checkbox'); },
-        );
-        it('should not be checked', () => { itBehavesLikeNotChecked(input); });
+      it('should have type checkbox', () => {
+        expect(input.attributes('type')).toBe('checkbox');
       });
 
-      describe('Checkbox Label Tests', () => {
-        it('should exist', () => { expect(label.exists()).toBe(true); });
-        it(
-          'should match provided label prop',
-          () => { expect(label.text()).toBe(propsData.label); },
-        );
+      it('should not be checked', () => {
+        expect(input.element.checked).toBe(false);
       });
     });
 
-    describe('When a label prop is provided', () => {
-      // Test Setup
-      beforeEach(() => {
-        propsData = { ...basePropsData, label: 'My Label' };
-        _setWrappers();
+    describe('Checkbox Label Tests', () => {
+      it('should exist', () => {
+        expect(label.exists()).toBe(true);
       });
 
-      it('should exist', () => { expect(label.exists()).toBe(true); });
+      it('should match provided label prop', () => {
+        expect(label.text()).toBe(baseProps.label);
+      });
     });
 
-    describe('When a description prop is provided', () => {
-      // Test Setup
-      beforeEach(() => {
-        propsData = { ...basePropsData, description: 'Description' };
-        _setWrappers();
+    describe('When a label prop is provided', () => {
+      it('should exist', () => {
+        mockProps = { label: 'My Label' };
+
+        updateWrapper();
+
+        expect(label.exists()).toBe(true);
       });
+    });
```

Comment:
1. Shouldn't this be inside `describe('Checkbox Label Tests')` ?
2. This is not actually testing that the passed label prop is equal to the label text. The label always exists as it is part of `baseProps`

Reply:
Im not doing a review of the test, just rewriting it with the new template. 

And I will not do a review of the test, it is not part of the ticket. I can do some modifications or improvements if I catch it but definitely not check if it is well asserted.

This new way...

```
describe('When a label prop is provided', () => {
      it('should exist', () => {
        mockProps = { label: 'My Label' };

        updateWrapper();

        expect(label.exists()).toBe(true);
      });
    });
```

Is the same as this old way...

```
describe('When a label prop is provided', () => {
      // Test Setup
      beforeEach(() => {
        propsData = { ...basePropsData, label: 'My Label' };
        _setWrappers();
      });

      it('should exist', () => { expect(label.exists()).toBe(true); });
    });
```

And regarding what you concern, yes seems like this test it is not fine...
As I can see in the `describe('When a label prop is provided'`

Seems like this code `propsData = { ...basePropsData, label: 'My Label' };` is not necessary and you can assert if the label exist directly because its already provided by the baseProps

So the new updated code should be 

```
describe('When a label prop is provided', () => {
      it('should exist', () => {
        expect(label.exists()).toBe(true);
      });
    });
```

Reply:
Cool, if it's out of the scope of this PR, this is ok.